src/modules/navigator/mission_block.cpp

```diff
@@ -109,8 +115,8 @@ MissionBlock::is_mission_item_reached_or_completed()
 	case NAV_CMD_DO_VTOL_TRANSITION:
 
 		if (int(_mission_item.params[0]) == 3) {
-			// transition to RW requested, only accept waypoint if vehicle state has changed accordingly
-			return _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING;
+			return _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+			       && !_navigator->get_vstatus()->in_transition_mode;
```

Comment:
Why was this change required? Didn't we on purpose have it accept the transition WP already when the transition was started before?
And I find the comment "transition to RW requested etc" useful, would you mind putting it back? Otherwise one has to look up what "3" stands for.

Reply:
@sfuhrer Yes, that was the logic before, but I don't it makes sense. Why do you accept a transition when the transition is not done. That's exactly what causes the position controllers and the navigator to be out of sync. Navigator will happily move to the next waypoint while the controller is still dealing with a vehicle in full fw condition.
I am not saying you can't make it work, but I'd rather have everything in sync as good as possible.

Commands are a bit weird in that case.